server/graphql/v1/mutations/collectives.js

```diff
@@ -377,12 +378,26 @@ export function editCollective(_, args, req) {
         return Promise.reject(new errors.Unauthorized({ message: errorMsg }));
       }
     })
-    .then(() => {
+    .then(async () => {
       // If we try to change the host
       if (
         newCollectiveData.HostCollectiveId !== undefined &&
         newCollectiveData.HostCollectiveId !== collective.HostCollectiveId
       ) {
+        if (newCollectiveData.HostCollectiveId) {
+          const host = await models.Collective.findByPk(newCollectiveData.HostCollectiveId);
+          const hostPlan = await host.getPlan();
+
+          if (hostPlan.hostedCollectivesLimit && hostPlan.hostedCollectivesLimit <= hostPlan.hostedCollectives) {
+            notifyAdminsOfCollective(host.id, {
+              type: 'hostedCollectives.otherPlans.limit.reached',
+              data: { name: host.name },
+            });
+            throw new errors.PlanLimit({
+              message: `This host, ${host.name}, has reached the maximum number of Collectives for their plan on Open Collective. They need to upgrade to a new plan to host your collective.`,
+            });
+          }
+        }
         return collective.changeHost(newCollectiveData.HostCollectiveId, req.remoteUser);
```

Comment:
`changeHost` is calling `addHost` and we already have some handling there, let's not duplicate.